src/test/java/org/opensearch/neuralsearch/processor/combination/GeometricMeanScoreCombinationTechniqueTests.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.neuralsearch.processor.combination;
+
+import static org.opensearch.neuralsearch.processor.combination.ArithmeticMeanScoreCombinationTechnique.PARAM_NAME_WEIGHTS;
+
+import java.util.List;
+import java.util.Map;
+
+public class GeometricMeanScoreCombinationTechniqueTests extends BaseScoreCombinationTechniqueTests {
+
+    private ScoreCombinationUtil scoreCombinationUtil = new ScoreCombinationUtil();
+
+    public GeometricMeanScoreCombinationTechniqueTests() {
+        this.expectedScoreFunction = this::geometricMean;
+    }
+
+    public void testLogic_whenAllScoresPresentAndNoWeights_thenCorrectScores() {
+        ScoreCombinationTechnique technique = new GeometricMeanScoreCombinationTechnique(Map.of(), scoreCombinationUtil);
+        testLogic_whenAllScoresPresentAndNoWeights_thenCorrectScores(technique);
+    }
+
+    public void testLogic_whenNotAllScoresPresentAndNoWeights_thenCorrectScores() {
+        ScoreCombinationTechnique technique = new GeometricMeanScoreCombinationTechnique(Map.of(), scoreCombinationUtil);
+        testLogic_whenNotAllScoresPresentAndNoWeights_thenCorrectScores(technique);
+    }
+
+    public void testLogic_whenAllScoresAndWeightsPresent_thenCorrectScores() {
+        List<Double> weights = List.of(0.9, 0.2, 0.7);
+        ScoreCombinationTechnique technique = new GeometricMeanScoreCombinationTechnique(
+            Map.of(PARAM_NAME_WEIGHTS, weights),
+            scoreCombinationUtil
+        );
+        testLogic_whenAllScoresAndWeightsPresent_thenCorrectScores(technique, weights);
+    }
+
+    public void testLogic_whenNotAllScoresAndWeightsPresent_thenCorrectScores() {
+        List<Double> weights = List.of(0.9, 0.2, 0.7);
+        ScoreCombinationTechnique technique = new GeometricMeanScoreCombinationTechnique(
+            Map.of(PARAM_NAME_WEIGHTS, weights),
+            scoreCombinationUtil
+        );
+        testLogic_whenNotAllScoresAndWeightsPresent_thenCorrectScores(technique, weights);
+    }
+
+    private float geometricMean(List<Float> scores, List<Double> weights) {
```

Comment:
Hope this comment is not too late. I should have paid attention to test code before.

Here, we are coping calculation logic from the production code which might not be able to catch a bug in the logic because the test code and production code is same.
Instead, we should calculate the end result manually and use it to compare withe the value returned by method. Same for other tests.

Reply:
I think we have two cases here: for the dimple dataset size, something like 2 sub-queries, 4-5 scores each we can calculate that manually and use it as expected value in assertion. For the larger size data manual calculation may not be feasible, so we need to calculate that ground truth value of expected values. Something to what we're doing in [knn repo tests](https://github.com/opensearch-project/k-NN/blob/main/src/testFixtures/java/org/opensearch/knn/TestUtils.java#L196). Let me rework into those two approaches.

Reply:
You can use the value returned by the method to get the initial value and hard code the value here so that if the logic changes in the future, we can catch a mistake.